### PR TITLE
RC #183

### PR DESCRIPTION
**Pull request by jason-capsule42**
# Alaska Airlines Pull Request

Release candidate pull request. See issue #183 for details.

## Review Checklist:

- [ ] My update follows the CONTRIBUTING guidelines of this project
- [ ] I have performed a self-review of my own update

<details open>
<summary>
RC Checklist
</summary>

### Browsers

[Browsers Support Guide](https://auro.alaskaair.com/support/browsersSupport)

[Dev demo link](https://alaskaairlines.github.io/auro-flight/)

Android

- [x] Chrome
- [x] Firefox

 iOS

- [x] Chrome
- [x] Firefox
- [x] Safari

Desktop

- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Edge

### Scenarios

- [ ] Validated linked issues with issue reporting team
- [ ] Test coverage report review

[Framework playground ](https://github.com/AlaskaAirlines/auro-framework-playground)

- [ ] Next React
- [ ] SvelteKit

</details><br>
**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Improve flight-card accessibility with localized screen-reader announcements, robust dynamic behavior, and updated component compatibility.

New Features:
- Provide complete screen-reader announcements for flight cards, including routes, times, flight numbers, stops, reroutes, cancellations, and duration.
- Add customizable internationalization templates and consumer-controlled aria-label overrides for accessibility announcements.

Bug Fixes:
- Correct ISO timestamp handling, duration formatting, flight-number spacing, cancellation updates, observer reconnection, and null-safe flight data handling.
- Update flight rendering for the auro-datetime v5 API and resolve Custom Elements Manifest typing and attribute contract issues.

Enhancements:
- Consolidate accessibility content into a single coherent flight-card announcement and reliably hide visual flight details from assistive technologies.

Build:
- Upgrade Auro development, component, styling, and design-token dependencies.

Documentation:
- Document accessibility and localization attributes, aria-label overrides, and updated component API details with new usage examples.

Tests:
- Expand coverage for accessibility announcements, localization, dynamic updates, edge cases, utility functions, and component behavior.

Chores:
- Update ownership metadata and generated README formatting.

### CODE REVIEW

File: src/auro-flight.js [hunk 1]
```diff
@@ -79,49 +109,351 @@ export class AuroFlight extends LitElement {
...
+
+    const flights = this.flights ?? [];
+    const flightId =
+      flights.length === 1
+        ? `Flight ${Array.from(flights[0])
+            .filter((c) => c !== " ")
+            .join(" ")}`
+        : flights.length === 0
+          ? ""
+          : "Multiple flights";
```
Comment (blackfalcon) [inline src/auro-flight.js]: Gap worth closing: the `flightId` strings are still hard-coded English. "Flight " (prefix) and "Multiple flights, a user who localizes all fifteen `i18n-*` attributes still gets an English “Flight A S 1 2 3 4” or “Multiple flights” at the front of every announcement. 

Recommendation: i18n-flight ("Flight {number}") and i18n-multiple-flights attributes.

Comment (blackfalcon) [inline src/auro-flight.js]: Really solid i18n coverage across the 15 attributes — one gap worth closing while we're here: the `flightId` strings are still hard-coded English. `"Flight "` (prefix) and `"Multiple flights"` never pass through `interpolate()`, so a consumer who localizes all fifteen `i18n-*` attributes still gets an English "Flight A S 1 2 3 4" or "Multiple flights" at the front of every announcement. Suggest `i18n-flight` (`"Flight {number}"`) and `i18n-multiple-flights` attributes to finish the story. Fine as a fast-follow if you'd rather not touch the RC — but worth an issue before a partner files it for us.

File: src/auro-flight.js [hunk 2]
```diff
@@ -150,16 +520,19 @@ export class AuroFlight extends LitElement {
...
   }
 
-  // function that renders the HTML and CSS into  the scope of the component
   render() {
+    const label = this._ariaLabelOverride ?? this._buildAriaLabel();
     return html`
       <section part="flightContainer">
+        <!-- Zero-width space keeps the element non-empty so Safari VoiceOver doesn't skip it;
+             aria-label provides the announcement without VoiceOver pronouncing punctuation as "full stop". -->
+        <span role="text" class="sr-label" aria-label="${label}">&#8203;</span>
```
Comment (blackfalcon) [inline src/auro-flight.js]: Docs note, not a blocker: the component correctly leaves `lang` to the host page rather than setting it on inner elements. One line in the API docs would close the loop: "if your `i18n-*` strings differ from the page language, set `lang` on `auro-flight` or an ancestor so AT picks the matching voice." Cheap insurance against a confusing AT bug report on mixed-language pages.

File: src/auro-flight.js [hunk 3]
```diff
@@ -15,62 +17,90 @@ import "@aurodesignsystem/auro-flightline";
...
+});
+
 // See https://git.io/JJ6SJ for "How to document your components using JSDoc"
 /**
  * The `auro-flight` element renders a DoT compliant Flight listing.
  * @customElement auro-flight
- * 
+ *
  * This design has been tested via the Alaska Legal team for legal compliance.
  * Please DO NOT modify unit tests pertaining to DoT regulations.
```
Comment (blackfalcon) [inline src/auro-flight.js]: Optional, while we're in this file: the "tested via the Alaska Legal team… DO NOT modify unit tests pertaining to DoT regulations" JSDoc dates to the 2021 initial commit (9642a49) and originally ended "without contacting gus@alaskaair.com" — the contact got stripped somewhere along the way. As written it reads like a live Legal control but points at nothing. Suggest either dropping it or rewording to whatever's actually current, so future reviewers don't burn time on it.

File: src/auro-flight.js [hunk 4]
```diff
@@ -79,49 +109,351 @@ export class AuroFlight extends LitElement {
...
+        } else {
+          key = isLast
+            ? this.i18nLastLayoverNoDuration
+            : this.i18nLayoverNoDuration;
+        }
+        let stopText = interpolate(key, {
+          station,
+          duration: segment.duration ?? "",
+        });
+        if (segment.canceled) stopText += ` ${this.i18nCanceled}`;
```
Comment (blackfalcon) [inline src/auro-flight.js]: Tiny one: a canceled stop announces as "with a stop in SEA canceled" — the `i18nCanceled` token is appended with a bare space, which reads awkwardly in English and will compose even worse in languages with different word order. Consider a template instead (`i18n-canceled-stop`, default `"{stop}, canceled"`) so translators control placement. Cosmetic; fast-follow is fine.

**Review by blackfalcon — APPROVED** [review]
This was AI overstepping ...

File: src/auro-flight.js (hunk 4 above)
Comment (blackfalcon) [inline src/auro-flight.js]: This is a little nit-pick, but there is something that may make this better. This line is the only one not using `interpolate`, so the string that comes from this will simply be "with a stop in SEA canceled" with SEA being read back as "sea". 

I'd recommend that we update line 39 to be `i18nCanceled: "{text}, canceled",`

Then, with line 411 update that to 
`if (segment.canceled) stopText = interpolate(this.i18nCanceled, { text: stopText });` which would give us a proper string of `"with a layover in S E A for 2h 10m, canceled"`

This gives us a properly formatted string that has a proper pause. 

This would also apply to line #421.

**Comment by jason-capsule42** [comment]
> One pre-ship condition I'd like explicit ownership on: the RC checklist browser/AT matrix is unchecked and the post-mortem itself lists TalkBack/NVDA/JAWS verification as pending. The approach leans on `role="text"`, which is WebKit-only — the Windows/Android AT pass is exactly the evidence that de-risks it, and I'd sequence that sign-off _before_ ship, not after. @jason-capsule42 — who owns that pass, and does this RC wait for it?

@sun-mota It would be good to update the post-mortem to call out that only Mac/iOS/Android tests are/were able to be run for this since we don't have Windows machine access at this time.

@blackfalcon The post-mortems are authored by the implementing engineer and intentionally only capture the coding work executed, not testing work the engineer has done. Though they typically call out testing scenarios that need to be done. If testing finds issues during development and code changes are made as a result, that *does* get captured in the post-mortem since it results in coding work.

But any actual "this test passes" sort of reporting is the responsibility of
A) automated test results reflected in the github actions
B) the person who actually is executing and submitting a review - the expectations during review is that the reviewer is validating all of the tickets acceptance criteria, reviewing the post-mortem and PR/commit content, and then testing everything outlined within as well as any other testing they may deem necessary. Only then is it marked "Approved".

We are working through options to define manual testing plans for components (which would get updated as additional work goes on over time) and then trying to solve how do we capture those results. Right now we don't have a good formal, recorded tracking method. We are considering options such as ADO supports test cases/suites/plans.

File: docs/post-mortem/1499242.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+- `_ariaLabelOverride` was declared `{ state: true }` without `attribute: false`. Lit treats `state: true` as equivalent to `attribute: false` at runtime, but CEM does not recognize this equivalence and still emitted the private property in the element's public attributes list, leaking it into editor autocomplete as a settable attribute.
+
+---
+
+## The Fix
+
+**Screen reader announcement** — A visually-hidden `<span role="text" class="sr-label" aria-label="${label}">&#8203;</span>` is the first child of the shadow DOM `<section>`. `_buildAriaLabel()` assembles the full announcement from flight number, departure, arrival, day diff, stops/layovers, reroutes, nonstop/canceled, and duration.
+
+**VoiceOver anchor** — `&#8203;` (zero-width space) satisfies Safari's non-empty text requirement without producing spoken output. `role="text"` (WebKit extension) forces VoiceOver to read `aria-label` rather than text content, preventing punctuation from being announced as "full stop." `.sr-label` is sized `100% × 100%` so VoiceOver's focus rectangle covers the full card; `pointer-events: none` prevents click interception.
```
Comment (blackfalcon) [inline docs/post-mortem/1499242.md]:
```
`role="text"` (WebKit extension) forces VoiceOver to read `aria-label` rather than text content
```

Its primary job is to tell VoiceOver to treat a group of mixed inline elements (e.g., text broken up by `<span>`, `<b>`, ``, etc.) as a single, unified string for navigation purposes, rather than stopping at each child element as a separate stop. 

I can see where the confusion comes in with `aria-label`. A `span` is role-generic, and an `aria-label` on that element is ignored. Using `role="text"` doesn't force the reading of `aria-label` content; it permits it to be read. 

I just want to be sure that we are not misrepresenting something here.

**Review by blackfalcon — DISMISSED** [review]
Strong work — and the post-mortem (docs/post-mortem/1499242.md) deserves a callout. Thirteen root causes, the reasoning behind each fix, and the approaches that didn’t work with evidence. Anyone touching AT behavior in Auro should read it. I checked the code against its claims and it all holds up, with tests.

File: src/auro-flight.js (hunk 3 above)
Comment (jason-capsule42) [reply src/auro-flight.js]: @blackfalcon Looks like it was actually you that removed it back in 2022.

The reference to gus@alaskaair.com in the JSDoc of [src/auro-flight.js](vscode-webview://03fttbe88tddn2dic8k2dne3e7kl0qdf5258kbfps8opbfgrc87p/src/auro-flight.js) was removed in commit 5c0abf6:

Commit: 5c0abf6fe7b99264be22ae31a674df0f60b88cb5 — "docs(api): update docs and include all elements into API document"
Author: Dale Sande <dale.sande@gmail.com>
Date: Mon Jun 27, 2022
The exact change was a one-line edit to the JSDoc:


- * Please DO NOT modify unit tests pertaining to DoT regulations without contacting gus@alaskaair.com.
+ * Please DO NOT modify unit tests pertaining to DoT regulations.
Confirming the reviewer's account
The reviewer (blackfalcon) was right about the origin: the line with the gus@alaskaair.com contact was present in the initial commit 9642a49 (2021) and survived until this June 2022 docs-cleanup commit stripped the "without contacting [gus@alaskaair.com](mailto:gus@alaskaair.com)" clause — leaving the "live Legal control that points at nothing" that they flagged.

A pickaxe search (git log -S "gus@alaskaair.com") also surfaces 31118e4 ("style: pixel pushing review", Feb 2021), but that's a false lead for the JSDoc — it added the contact to a different file (the docs/README), and the pickaxe also matched it partly because that commit was authored by Gus Naughton himself. The JSDoc in src/auro-flight.js was only ever touched by 9642a49 (added) and 5c0abf6 (removed).

Comment (blackfalcon) [reply src/auro-flight.js]: This was a finding from AI that was reviewed and dismissed. Should not be in this PR review.

**Comment by sun-mota** [comment]
https://itsals.visualstudio.com/E_Retain_Content/_workitems/edit/1637686
ADO ticket created

**Review by sun-mota — APPROVED** [review]
